File: benchmark/results/python-invocation-script/run-004/a-mcp/output/shipping/services/carrier_gateway.py
from shipping.aggregates.shipment import Shipment

"""CarrierGateway service."""
class CarrierGateway:
    def create(self, input: dict) -> Shipment:
        raise NotImplementedError('not implemented')
    def find_by_id(self, id: str) -> Shipment | None:
        raise NotImplementedError('not implemented')
    def list(self, limit: int) -> list[Shipment]:
        raise NotImplementedError('not implemented')
    def delete(self, id: str) -> None:
        raise NotImplementedError('not implemented')

File: benchmark/results/java-multilang/run-004/b-baseline/output/com/metaengine/demo/shipping/services/CarrierGateway.java
package com.metaengine.demo.shipping.services;

import java.util.List;

import com.metaengine.demo.shipping.aggregates.Shipment;

/** CarrierGateway integration service. */
public class CarrierGateway {

    /** Submit a Shipment to the carrier. */
    public Shipment create(Shipment input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Find a Shipment by id (may return null). */
    public Shipment findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** List shipments up to limit. */
    public List<Shipment> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Delete a Shipment by id. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}
